src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -548,7 +548,7 @@ def resolve_imports_and_packages(
     statement_params: Optional[Dict[str, str]] = None,
     source_code_display: bool = False,
     skip_upload_on_content_match: bool = False,
-    force_push: bool = True,
+    force_push: bool = False,
 ) -> Tuple[str, str, str, str, str, bool]:
```

Comment:
By default, force_push will be False. The user needs to explicitly confirm that they wish to upload packages which depend on native code.